coding-questions-python/longest-palindromic-substring.py

```diff
@@ -0,0 +1,74 @@
+'''
+Given a string, find the longest substring which is palindrome. For example, if the given string is “forgeeksskeegfor”, the output should be “geeksskeeg”.
+
+Suggested approach:
+Dynamic programming -- maintain a 2D array that stores a boolean at each index
+	- if inex i,j is true, that means that the substring from index i to j (inclusive) is a palindrome
+This way, we can use pre-computed results of smaller lengths to find the current result for the (current) lengths
+'''
+
+
+def init():
+	print(dummy)
+
+def longestPalindromicSubstring(str):
+	cache = []
+	longestLength = -1
+	longestSubstring = ""
+
+	
+	#init 2D array with False at all indices
+	for i in range(len(str)):
```

Comment:
Create matrix in python using list comprehensions
`cache = [[False for _ in str] for _ in str]`
https://stackoverflow.com/questions/6667201/how-to-define-a-two-dimensional-array-in-python